pennylane/optimize/gradient_descent.py

```diff
@@ -47,86 +48,116 @@ def update_stepsize(self, stepsize):
             grad_fn (function): Optional gradient function of the
                 objective function with respect to the variables ``x``.
                 If ``None``, the gradient function is computed automatically.
+                Must match shape of autograd derivative.
+            **kwargs : Variable length of keywords for the objective function
 
         Returns:
-            array: the new variable values :math:`x^{(t+1)}`
+            list: the new variable values :math:`x^{(t+1)}`
         """
-        g, _ = self.compute_grad(objective_fn, x, grad_fn=grad_fn)
-        x_out = self.apply_grad(g, x)
 
-        return x_out
+        g, _ = self.compute_grad(objective_fn, args, kwargs, grad_fn=grad_fn)
+        new_args = self.apply_grad(g, args)
+
+        # unwrap from list if one argument, cleaner return
+        if len(new_args) == 1:
+            return new_args[0]
+
+        return new_args
 
     @staticmethod
-    def compute_grad(objective_fn, x, grad_fn=None):
+    def compute_grad(objective_fn, args, kwargs, grad_fn=None):
         r"""Compute gradient of the objective_fn at the point x and return it along with the
             objective function forward pass (if available).
 
         Args:
             objective_fn (function): the objective function for optimization
-            x (array): NumPy array containing the current values of the variables to be updated
+            args (tuple): Tuple of NumPy arrays containing the current parameters for the
+                objection function
+            kwargs (dict): Keywords for the objective function
             grad_fn (function): Optional gradient function of the objective function with respect to
                 the variables ``x``. If ``None``, the gradient function is computed automatically.
+                Must return same shaped tuple(array) as autograd derivative
 
         Returns:
-            tuple: The NumPy array containing the gradient :math:`\nabla f(x^{(t)})` and the
+            tuple(array): The NumPy array containing the gradient :math:`\nabla f(x^{(t)})` and the
                 objective function output. If ``grad_fn`` is provided, the objective function
                 will not be evaluted and instead ``None`` will be returned.
         """
         g = get_gradient(objective_fn) if grad_fn is None else grad_fn
-        grad = g(x)
+        grad = g(*args, **kwargs)
         forward = getattr(g, "forward", None)
 
         return grad, forward
 
-    def apply_grad(self, grad, x):
-        r"""Update the variables x to take a single optimization step. Flattens and unflattens
+    def apply_grad(self, grad, args):
+        r"""Update the variables to take a single optimization step. Flattens and unflattens
         the inputs to maintain nested iterables as the parameters of the optimization.
 
         Args:
-            grad (array): The gradient of the objective
+            grad (tuple(array)): The gradient of the objective
```

Comment:
I just looked up various standards for this case. Seems like (tuple of array) is a recommendation.

Reply:
I just followed how we've written things in other parts of PennyLane. I think square brackets have been used rather than parenthesis (might be good to double check to make sure we just follow the same conventions). 🙂